doc/dev/developer_guide.md

```diff
@@ -36,6 +39,77 @@ $ make test
 For more information on running testing and correctly configuring your environment,
 refer to the [`Running the Tests Locally`][running-the-tests] document.
 
+To run the lint checks done in the CI locally, run:
+
+```sh
+$ make lint
+```
+
```

Comment:
**NOTE** Note that for it is required to install `golangci-lint` locally. For more info see its [doc](https://github.com/golangci/golangci-lint#install)